### PR TITLE
feat(cts): add e2e tests

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1791]

### Changes included:

- add e2e tests support
- enable formatter rules
- fix qp encoding
- add qp key encoding
- remove Builder protocol as it's not compatible with Linux (it triggers encoding-computing on the other URLComponents' properties)

[DI-1791]: https://algolia.atlassian.net/browse/DI-1791?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Fluf22** [comment]
lol wtf go failed

**Comment by shortcuts** [comment]
> lol wtf go failed

ah it seems flaky

**Comment by Fluf22** [comment]
Finally found the underlying issue for Swift
Go still in the choux, though

File: templates/swift/tests/requests/requests.mustache [hunk 1]
```diff
@@ -65,26 +102,43 @@ final class {{client}}RequestsTests: XCTestCase {
...
             let echoResponseHeaders = try XCTUnwrap(echoResponse.headers)
-            for header in comparableHeadersMap {
+            for header in expectedHeadersMap {
               XCTAssertEqual(echoResponseHeaders[header.key.capitalized], header.value)
             }
           {{/headers}}
         {{/request}}
+
+        {{#response}}
+          let e2eClient = try {{client}}(appID: APPLICATION_ID, apiKey: API_KEY{{#hasRegionalHost}}, region: .{{defaultRegion}}{{/hasRegionalHost}})
```
Comment (millotp) [inline templates/swift/tests/requests/requests.mustache:127]: can you create the client once for all the tests, to check if it works fine in parallel ?

**Review by millotp — COMMENTED** [review]
looks great ! small comment

File: templates/swift/tests/requests/requests.mustache (hunk 1 above)
Comment (Fluf22) [reply templates/swift/tests/requests/requests.mustache:127]: Yes

**Review by millotp — APPROVED** [review]
perfect !

**Comment by Fluf22** [comment]
Yay, let's go!

File: snippets/swift/.swiftformat
Comment (shortcuts) [inline snippets/swift/.swiftformat:1]: If we have the same rule for client and snippet we can maybe generate it

**Review by shortcuts — COMMENTED** [review]
Wonderful

File: snippets/swift/.swiftformat
Comment (Fluf22) [reply snippets/swift/.swiftformat:1]: Yeah, I'll do that in another PR!
I also want to put the swift version in it with the generator, and test some other rule options ☺️

**Pull request merged**